iocore/aio/AIO.cc

```diff
@@ -379,18 +379,15 @@ aio_queue_req(AIOCallbackInternal *op, int fromAPI = 0)
 static inline int
 cache_op(AIOCallbackInternal *op)
 {
-  bool read = (op->aiocb.aio_lio_opcode == LIO_READ);
+  using ActionFunc = decltype(&pread);
+  ActionFunc action = op->aiocb.aio_lio_opcode == LIO_READ ? &pread : reinterpret_cast<ActionFunc>(&pwrite);
```

Comment:
I am sorry, but I don't think the code is an improvement.  It has a decltype and a reinterpret_cast to save a bool conditional before doing a syscall.  It isn't going to effect performance.

Also, it now makes debugging harder since you can't break easily on the read or write.

Reply:
I know that this is a negligible optimization. Actually, I have not encountered a case where the loop was repeated more than 4 times. 
This is an part of a new functionality at the Cache module. I prefer push a independent part of a code, before main commit, to simplification and reduce size of a main commit.